src/evict/evict_lru.c

```diff
@@ -891,17 +880,63 @@ __evict_clear_walk(WT_SESSION_IMPL *session)
      */
     btree->evict_ref = NULL;
 
+    if (clear_pos) {
+        btree->evict_pos = WT_NPOS_INVALID;
+        btree->evict_ref_saved = NULL;
+    } else {
+        /*
+         * Remember the last position before clearing it so that we can restart from about the same
+         * point later.
+         */
+
+        /*
+         * If we're at an internal page, then we've just finished all its leafs, so get the position
+         * of the very beginning or the very end of it depending on the direction of walk. For leaf
+         * pages, use the middle of the page.
+         */
+        btree->evict_ref_saved = ref;
+        if (!WT_VERBOSE_LEVEL_ISSET(session, WT_VERB_EVICTION, WT_VERBOSE_DEBUG_1)) {
```

Comment:
Duplicating the position set logic in order to have a "where" variable isn't ideal. I would move the pos variable setting outside the if statement and create a __debug_npos_to_where_string function:
```
       btree->evict_ref_saved = ref;
        pos = F_ISSET(ref, WT_REF_FLAG_LEAF) ? WT_NPOS_MID :
              btree->evict_start_type == WT_EVICT_WALK_NEXT ||
                btree->evict_start_type == WT_EVICT_WALK_RAND_NEXT ?
                                                   WT_NPOS_RIGHT :
                                                   WT_NPOS_LEFT;
        if (!WT_VERBOSE_LEVEL_ISSET(session, WT_VERB_EVICTION, WT_VERBOSE_DEBUG_1))
            btree->evict_pos = __wt_page_npos(session, ref, pos, NULL, NULL, 0);
        else {
            btree->evict_pos =
              __wt_page_npos(session, ref, pos, path_str, &path_str_offset, PATH_STR_MAX);
            __wt_verbose_debug1(session, WT_VERB_EVICTION,
              "Evict walk point memorized at position %lf %s of %s page %s ref %p",
              btree->evict_pos, __debug_npos_to_where_string(pos), F_ISSET(ref, WT_REF_FLAG_INTERNAL) ? "INTERNAL" : "LEAF",
              path_str, (void *)ref);
        }
```

Reply:
I made some changes